usaspending_api/common/api_request_utils.py

```diff
@@ -219,6 +219,33 @@ def range_intersect(self, fields, values):
         return Q(**q_case)
 
 
+# Handles unique value requests
+class UniqueValueHandler():
+    @staticmethod
+    # Data set to use (should be filtered already)
+    # Fields to find unique values for
+    # Returns a dictionary containing fields, values and their counts
+    # {
+    #   "recipient__name": {
+    #       "Jon": 5,
+    #       "Joe": 2
+    #   }
+    # }
+    def get_values_and_counts(data_set, fields):
+        data_set = data_set.all()  # Do this because we don't want to get finnicky with annotations
+        response_object = {}
+        if fields:
+            for field in fields:
+                response_object[field] = {}
+                unique_values = data_set.values(field).distinct()
+                for value in unique_values:
+                    q_kwargs = {}
+                    q_kwargs[field] = value[field]
+                    print(q_kwargs)
```

Comment:
Ach! I'm embarrassed, I usually get all those cleared out. I guess I was excited to be done.